backend/dal/Helpers/Extensions/BuildingExtensions.cs

```diff
@@ -24,7 +24,11 @@ public static class BuildingExtensions
         /// <exception type="DbUpdateException">The name within a parcel should be unique.</exception>
         public static void ThrowIfNotUnique(this PimsContext context, Entity.Parcel parcel, Entity.Building building)
         {
-            var parcelBuildings = context.Parcels.Where(p => p.Id == parcel.Id).SelectMany(p => p.Buildings.Select(b => b.Building.Name)).Distinct().ToArray();
+            if(parcel == null || String.IsNullOrEmpty(building.Name))
```

Comment:
If it's not named or belongs to a parcel don't worry about being unique?

Reply:
Yes, unfortunately we need this as there are parcels with buildings that already violate this rule within the DB. For example
Parcel 1
  Building 1 Name: ''
  Building 2 Name: ''
  Building 3 Name: 'Test'
If I modify Building 3, this check was previously throwing an exception.